app/models/manageiq/providers/openstack/network_manager/network_router.rb

```diff
@@ -30,6 +30,280 @@ class ManageIQ::Providers::Openstack::NetworkManager::NetworkRouter < ::NetworkR
 
   supports :remove_interface
 
+  def self.params_for_create(ems, cloud_network_id)
+    if cloud_network_id.nil?
```

Comment:
Interesting that the form is completely different depending on if a cloud_network_id is passed it...but then [[down here]](https://github.com/ManageIQ/manageiq-providers-openstack/pull/731/files#diff-493079fa00c2f1ecbd1a2b59f7c9fb859429e054695f9c06501577c3e537ed7eR129) we have to select a cloud network anyway

Reply:
Yeah we end up calling this twice. Once with no cloud_network_id for the first part of the form and the second time for the last field of the form which requires a cloud_network_id to get the options for the subnets list.

Reply:
Could you make the last one a subform that depends on the selection of a cloud network?

Reply:
Yeah I can try

Reply:
Okay wonder if we should hold off on this and tackle it as part of the "dialog workflow" API that we've discussed (cc @Fryguy @kavyanekkalapu)

Reply:
darn. most or our options only take an `ems` here and not an additional parameter.
In a perfect world, this would be consistent across all models (though we do not live in that world)

I do feel options in general should not take an additional parameter. So that looks a little fishy to me